### PR TITLE
[core] Unify axis normalization/validation utils

### PR DESCRIPTION
**Pull request by praasz**
### Details:
 - Split function for smaller simper utils, responsible for validation or normalization or more complex doing both.
 - Unify the functions parameters order
 - Remove redundant check of rank
 - Produce smaller binary size
 - Fix Coverity issue `Improper use of negative value`.

### Tickets:
 - CVS-136544

### CODE REVIEW

**Comment by praasz** [comment]
@build_jenkins

**Comment by praasz** [comment]
build_jenkins

File: src/common/transformations/src/transformations/common_optimizations/transpose_sinking.cpp [hunk 1]
```diff
@@ -181,9 +181,10 @@ ov::pass::TransposeReduction::TransposeReduction() {
         if (!transpose_order || !reduction_axes)
             return false;
 
-        const auto& non_negative_axes = ov::util::normalize_axes(reduction->get_friendly_name(),
-                                                                 reduction_axes->cast_vector<int64_t>(),
-                                                                 reduction->get_input_partial_shape(0).rank());
+        const auto non_negative_axes =
```
Comment (barnasm1) [inline src/common/transformations/src/transformations/common_optimizations/transpose_sinking.cpp:184]: Consider to use const ref.

File: src/frontends/onnx/frontend/src/op/hardmax.cpp [hunk 2]
```diff
@@ -66,7 +66,8 @@ ov::OutputVector hardmax(const ov::frontend::onnx::Node& node) {
     const auto& input_shape = input.get_partial_shape();
 
     auto axis = node.get_attribute_value<std::int64_t>("axis", -1);
-    axis = ov::util::normalize_axis(node.get_description(), axis, input_shape.rank());
+    // TODO:2
```
Comment (barnasm1) [inline src/frontends/onnx/frontend/src/op/hardmax.cpp:69]: Is that comment needed?

File: src/frontends/onnx/frontend/src/op/scan.cpp [hunk 3]
```diff
@@ -50,10 +62,14 @@ ov::OutputVector scan_to_tensor_iterator(const ov::OutputVector& node_inputs,
         const auto axis_node = v0::Constant::create(ov::element::i64, ov::Shape{1}, {axis});
         auto shape = node_inputs[in_idx + in_offset].get_partial_shape();
         if (shape.rank().is_static()) {
-            axis = ov::util::normalize_axis(node_description,
-                                            scan_input_axes[i],
-                                            node_inputs[in_idx + in_offset].get_partial_shape().rank());
+            axis = try_normalize_axis_for_static_rank(axis, shape.rank());
+            // axis = common::normalize_axis(node_description, axis, shape.rank());
```
Comment (barnasm1) [inline src/frontends/onnx/frontend/src/op/scan.cpp:66]: remove comment

File: src/core/src/validation_util.cpp [hunk 4]
```diff
@@ -324,68 +318,55 @@ bool has_no_symbols(const ov::TensorSymbol& symbols) {
...
 }
 
-int64_t normalize_axis(const ov::Node* node,
-                       std::int64_t axis,
-                       std::uint64_t tensor_rank,
-                       std::int64_t axis_range_min,
-                       std::int64_t axis_range_max) {
-    return normalize_axis(node->description(), axis, tensor_rank, axis_range_min, axis_range_max);
+AxisVector try_get_normalized_axis_vector(const Tensor& tensor, const Rank& rank, const Node& node) {
+    auto axes_values = ov::get_tensor_data_as<int64_t>(tensor);
```
Comment (barnasm1) [inline src/core/src/validation_util.cpp:363]: Consider use `const auto& axes_values`

File: src/common/transformations/src/transformations/common_optimizations/transpose_sinking.cpp (hunk 1 above)
Comment (praasz) [reply src/common/transformations/src/transformations/common_optimizations/transpose_sinking.cpp:184]: Is not required as axes are created here.

File: src/frontends/onnx/frontend/src/op/hardmax.cpp (hunk 2 above)
Comment (praasz) [reply src/frontends/onnx/frontend/src/op/hardmax.cpp:69]: No, the coode should be updated here.

**Review by gkrivor — APPROVED** [review]
I was looking on a ONNX part, looks good for me

**Review by t-jankowski — APPROVED** [review]

File: src/core/src/validation_util.cpp [hunk 5]
```diff
@@ -324,68 +318,55 @@ bool has_no_symbols(const ov::TensorSymbol& symbols) {
...
+    try_normalize_axes(axes_values, rank, node);
+    return {axes_values.begin(), axes_values.end()};
```
[the remark this answers is not in the document]
Comment (praasz) [reply src/core/src/validation_util.cpp:365]: Current solution will give same result.

File: src/core/src/validation_util.cpp [hunk 6]
```diff
@@ -324,68 +318,55 @@ bool has_no_symbols(const ov::TensorSymbol& symbols) {
...
-                       std::uint64_t tensor_rank,
-                       std::int64_t axis_range_min,
-                       std::int64_t axis_range_max) {
-    // Accepted range of value for axis is [axis_range_min, axis_range_max].
-    OPENVINO_ASSERT((axis_range_min <= axis) && (axis <= axis_range_max),
-                    node_description,
-                    normalize_axis_error_msg(axis, axis_range_min, axis_range_max));
-    return normalize(axis, tensor_rank);
+AxisSet try_get_normalized_axis_set(const Tensor& tensor, const Rank& rank, const Node& node) {
+    return {try_get_normalized_axis_vector(tensor, rank, node)};
```
[the remark this answers is not in the document]
Comment (praasz) [reply src/core/src/validation_util.cpp:369]: Current solution will give same result.

**Pull request merged**